jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -0,0 +1,882 @@
+  {
+    return false;
+  }
+
+  if (property_desc_p->is_get_defined
+      || property_desc_p->is_set_defined)
+  {
+    return false;
+  }
+
+  if ((property_desc_p->is_configurable_defined && property_desc_p->is_configurable)
+      || (property_desc_p->is_enumerable_defined && !property_desc_p->is_enumerable)
+      || (property_desc_p->is_writable_defined && !property_desc_p->is_writable))
+  {
+    return false;
+  }
+
+  if (property_desc_p->is_value_defined)
+  {
+    return ecma_op_typedarray_set_number_prop (obj_p, num, property_desc_p->value);
+  }
+
+  return true;
+} /* ecma_op_typedarray_define_number_prop */
+
+/**
+ * Set the integer number property value of the typedarray
+ *
+ * See also: ES2015 9.4.5.9
+ *
+ * @return boolean, false if failed
+ */
+bool
+ecma_op_typedarray_set_number_prop (ecma_object_t *obj_p,/**< a TypedArray object */
+                                    ecma_number_t num, /**< the numeric index */
+                                    ecma_value_t value) /**< value of the property */
+{
+  JERRY_ASSERT (ecma_is_typedarray (ecma_make_object_value (obj_p)));
+
+  ecma_value_t error_val = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+
+  ECMA_OP_TO_NUMBER_TRY_CATCH (value_num, value, error_val);
+
+  int32_t index = ecma_number_to_int32 (num);
+  ecma_length_t array_length = ecma_typedarray_get_length (obj_p);
+
+  if (index < 0 || (ecma_length_t) index > array_length || num != ((ecma_number_t) index))
+  {
+    return ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+  }
+
+  ecma_typedarray_info_t *info_p = ecma_typedarray_get_info (obj_p);
+  ecma_length_t byte_pos = (ecma_length_t) index * info_p->bytes_per_element + info_p->byte_offset;
+  lit_magic_string_id_t class_id = ecma_object_get_class_name (obj_p);
+  lit_utf8_byte_t *target_p = ecma_arraybuffer_get_buffer (info_p->viewed_arraybuffer_p) + byte_pos;
+
+#define TYPEDARRAY_SET_INDEXED_ELEMENT(type) \
+  *((type *) target_p) = (type) value_num;
+
+  switch (class_id)
+  {
+    case LIT_MAGIC_STRING_INT8_ARRAY_UL:
+    {
+      TYPEDARRAY_SET_INDEXED_ELEMENT (int8_t);
+      break;
+    }
+    case LIT_MAGIC_STRING_UINT8_ARRAY_UL:
+    {
+      TYPEDARRAY_SET_INDEXED_ELEMENT (uint8_t);
+      break;
+    }
+    case LIT_MAGIC_STRING_UINT8_CLAMPED_ARRAY_UL:
+    {
+      uint8_t clamped;
+
+      if (value_num > 255)
+      {
+        clamped = 255;
+      }
+      else if (value_num < 0)
+      {
+        clamped = 0;
+      }
+      else
+      {
+        clamped = (uint8_t) value_num;
+
+        if (clamped + 0.5 < value_num
+            || (clamped + 0.5 == value_num && clamped % 2 == 1))
+        {
+          clamped ++;
+        }
+      }
+      *((uint8_t *) target_p) = clamped;
+      break;
+    }
+    case LIT_MAGIC_STRING_INT16_ARRAY_UL:
+    {
+      TYPEDARRAY_SET_INDEXED_ELEMENT (int16_t);
+      break;
+    }
+    case LIT_MAGIC_STRING_UINT16_ARRAY_UL:
```

Comment:
Endian problem is not a type cast problem, it is writing the bytes of a number in different order. This patch is big enough, and this is not needed at all for now.